src/Service/Controllers/RestController.cs

```diff
@@ -224,16 +224,21 @@ public static JsonResult ErrorResponse(string code, string message, HttpStatusCo
 
                 if (result is CreatedResult)
                 {
-                    // Location is made up of two parts, the first being constructed
-                    // from the HttpRequest found in the HttpContext. The other part
-                    // is the primary key route, which has already been saved in the
+                    // Location is made up of three parts, the first being constructed
+                    // from the HttpRequest found in the HttpContext. The second part being the
+                    // base route configured in the config file.
+                    // The third part is the primary key route, which has already been saved in the
```

Comment:
So the reason the locatation header logic is in Restcontroller is because you have no access to the runtime config (base path value) in SQLmutation engine? 

Did you consider keeping the location header population in sqlmutation engine, and just using this place in RestController to prepend the base path, if configured?

Reply:
1) Empty string is set only in case of stored procedures. For other entity types, non-empty value is returned. I've described the reason for returning an empty string for stored procedures in an earlier comment. I would say the comment is accurate. 

2) For non-stored procedures, the location is set with the PK of the newly created item. This is constructed in `SqlMutationEngine.ConstructPirmaryKeyRoute()`. The PK of the newly created item is accessible here, and hence, that calculation takes place here.
To extract the base-route, access to Runtime config is necessary, and therefore that logic is written in `RestService`. In `RestController`, all the three components - baseURL, base-route, and PK route are assembled to form the entire location header before returning the response.